software/last/lastal/main.nf

```diff
@@ -0,0 +1,48 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process LAST_LASTAL {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::last=1219" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/last:1219--h2e03b76_0"
+    } else {
+        container "quay.io/biocontainers/last:1219--h2e03b76_0"
+    }
+
+    input:
+    tuple val(meta), path(fastx)
+    path(index)
+    path(param_file)
+
+    output:
+    tuple val(meta), path("*.maf.gz"), emit: maf
+    path "*.version.txt"             , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def trained_params = param_file.exists() ? "-p ${param_file}"  : ""
+    """
+    INDEX_NAME=`ls lastdb/*bck | xargs basename -s .bck`
```

Comment:
Hmmm...is the database directory name hardcoded here to `lastdb/`? Are we able to make this more flexible to instances where you can use whatever name you like?

Reply:
Maybe using a `find` as we do here:
https://github.com/nf-core/modules/blob/4575e5455c62560e0cabdf128ec3adcc81f84a2c/software/bowtie2/align/main.nf#L38

Reply:
Thanks for spotting that I hardcoded the name.  I gave you a long answer in 8845ca9c4558b7986a263e512e6fd1c6b10326b4 for explaining why I stick using `basename`

Reply:
Awesome! Yes, we could give it a shot with `ls`. I read some stuff around the place about `ls` being inappropriate in some instances for this sort of things. If it breaks we can just fix it!